### PR TITLE
Fixed error checking with newer buffer API

### PR DESCRIPTION
**Pull request by alecl**
Fixes issue #4

### CODE REVIEW

**Comment by matthistuff** [comment]
Thanks for contributing. I have somewhat let this project sleep but will look into your changes as soon as possible.

**Comment by alecl** [comment]
Any news on merging this?

File: lib/buffer-api.js [hunk 1]
```diff
@@ -36,9 +36,14 @@ var util = require('util'),
 
           if (err) {
             return reject(err)
-          } else if (res.statusCode !== 200) {
+          } else if (res.statusCode !== 200 || (body && typeof(body.success) !== "undefined" && !body.success)) {
```
Comment (Kikobeats) [inline lib/buffer-api.js]: Can you extract the new condition into a named function?

For example:

```js
function hasInvalidResponse(body) {
...
}
```

Also first condition could be extracted as:

```js
function isHttpOkCode() {
...
}
```

More easy to maintain it in the future 😄

File: lib/buffer-api.js [hunk 2]
```diff
@@ -36,9 +36,14 @@ var util = require('util'),
...
             if (body) {
-              err = new Error(util.format(body.error))
+              if (body.error) {
```
Comment (Kikobeats) [inline lib/buffer-api.js]: better approach for that, determinate the input source for the method:

```js
let errorMessage
if (body) {
  if (body.error) errorMessage = body.error
  else errorMessage = body.message
}

const error = new Error(util.format(errorMessage))
```

It avoids repeat code

**Comment by Kikobeats** [comment]
I saw this now, review my comments and I can merge it 😄 

Thanks for your time, very appreciated PR 👌

**Comment by alecl** [comment]
seemed like a const was the way to go to check for a single numeric value so I added that. Plus it didn't seem like the rest of the code base was using let only var so did 

`var errorMessage` for consistency

Checked in changes.

**Comment by Kikobeats** [comment]
Released at `1.0.3`

**Pull request merged**